Fix building Wasm stdlib with macOS presets

MaxDesiatov opened the pull request:
Added optional `ignore_extra_cmake_options` to `cmake_product.py`, existing behavior is preserved with the default argument value set to `False`.

Passed `ignore_extra_cmake_options=True` in `wasisysroot.py`. This allows libc++ for WASI to be built correctly with `-fno-exceptions` in presence of `extra-cmake-options` in macOS presets (which are absent in Linux presets).

MaxDesiatov committed:
Build Wasm stdlib on macOS CI jobs

While tests are skipped, this prevents the job at least from regressing the build configuration on macOS, which is unfortunately quite frequent and disruptive for local development

MaxDesiatov commented:
@swift-ci smoke test

MaxDesiatov committed:
Fix `CMAKE_CXX_FLAGS` misconfiguration in `wasisysroot.py`

`CMAKE_CXX_FLAGS:STRING` is not the same as `CMAKE_CXX_FLAGS`. The former is ignored when building on macOS (some difference in CMake behavior, possibly in CMake versions?), while former is applied correctly. This means `-fno-exceptions` is passed correctly, when building libc++ for WASI.

MaxDesiatov committed:
Merge branch 'main' into maxd/wasm-stdlib-macos

MaxDesiatov committed:
Fix conflict resolution error in `wasisysroot.py`

MaxDesiatov commented:
@swift-ci build toolchain

kateinoigakukun commented on utils/swift_build_support/swift_build_support/products/wasisysroot.py:
```diff
@@ -195,8 +195,8 @@ def _build(self, host_target, enable_wasi_threads, compiler_rt_os_dir, target_tr
             cxx_flags.append('-pthread')
-        cmake.cmake_options.define('CMAKE_C_FLAGS:STRING', ' '.join(c_flags))
-        cmake.cmake_options.define('CMAKE_CXX_FLAGS:STRING', ' '.join(cxx_flags))
+        cmake.cmake_options.define('CMAKE_C_FLAGS', ' '.join(c_flags))
```
Could you elaborate on how the type specifier makes difference? Why is this specific to `CMAKE_C{,XX}_FLAGS`?

MaxDesiatov replied:
Disregard that, it's caused by `CMAKE_CXX_FLAGS` defined on macOS preset, which overrides these `CMAKE_CXX_FLAGS`, I'll clean that up.

MaxDesiatov committed:
Enable Wasm backend for macOS in `build-presets.ini`

MaxDesiatov commented:
@swift-ci smoke test macos

MaxDesiatov committed:
Revert "Fix `CMAKE_CXX_FLAGS` misconfiguration in `wasisysroot.py`"

MaxDesiatov committed:
Add optional `ignore_extra_cmake_options` to `cmake_product.py`

Existing behavior is preserved with the default argument value set to `False`.

MaxDesiatov committed:
Pass `ignore_extra_cmake_options=True` in `wasisysroot.py`

This allows libc++ for WASI to be built correctly with `-fno-exceptions` in presence of `extra-cmake-options` in macOS presets (which are absent in Linux presets).

MaxDesiatov committed:
Revert changes to `build-presets.ini`

It's premature to increase build time of existing macOS presets. We should fix the bug with macOS builds first, while modifications of existing presets should be out of scope of the change.
